src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -157,5 +161,13 @@ private static bool ValidateAzureArcEnvironment(string identityEndpoint, string
             logger?.Verbose(() => "[Managed Identity] Azure Arc managed identity is not available.");
             return false;
         }
+
+        internal void SetRuntimeMtlsBindingCertificate(X509Certificate2 cert)
+        {
+            var old = RuntimeMtlsBindingCertificate;
+            RuntimeMtlsBindingCertificate = cert;
+            //dispose prior stored cert on replacement
+            old?.Dispose();
```

Comment:
nit: I would swap these lines

Reply:
Added an XML remark clarifying that Dispose() releases the object’s resources and does not remove a cert from the OS store (that’s X509Store.Remove)